src/managers/RoleManager.js

```diff
@@ -188,6 +192,7 @@ class RoleManager extends CachedManager {
       hoist: data.hoist,
       permissions: typeof data.permissions === 'undefined' ? undefined : new Permissions(data.permissions),
       mentionable: data.mentionable,
+      icon: typeof data.icon === 'undefined' ? undefined : await DataResolver.resolveImage(data.icon),
```

Comment:
Is passing null any different from undefined here? If not you can just call the resolveImage function

Reply:
null would reset the icon most likely

Reply:
Then we can probably do this
```suggestion
      icon: (await DataResolver.resolveImage(data.icon)) ?? data.icon,
```

Reply:
Isn't that inconsistent then with how every other icon/image-taking methods do it?

Reply:
I’m not sure but this seems more optimal don’t you think?

Reply:
could so
```js
data.icon == undefined ? data.icon : ...
```
but honestly thats just a bit weird what is there now works fine

Reply:
I'm just trying to simplify the code and I think my suggestion works fine since resolveImage will return null if data.icon is falsy, and the nullish coalescing makes sure that the passed value is the value of data.icon in case this is falsy

Reply:
Adding it here will bring inconsistency. I think you should open another PR that adds this change throughout the library, so that it can be discussed and tracked at a single place.